### DISCUSSION TITLE
Looking for input on product violation severity definitions

### DISCUSSION THREAD

**Discussion by PaladinCloudJohn** [PaladinCloud/CE#1080] [Ideas]
We are working on documenting language around severity and auditing policies to be sure they are consistently evaluated. 

Paladin Cloud supports four severity levels. In decreasing order of severity, they are Critical, High, Medium, and Low. Paladin Cloud also supports four categories of violations: Security, Cost, Operations, and Tagging. 

Here are possible definitions for the four severity levels:

### Violation Severity Definitions
**Critical**

> Critical misconfigurations or vulnerabilities indicate something that is imminently exploitable or could lead to a complete compromise of your system or network. Critical violations are often actively exploited by attackers without any interaction required from the user and can result in significant damage or data loss. Therefore, a critical vulnerability should be addressed immediately and given top priority.

Example Critical Policy: [Deny Public Access to Non-allow Listed S3 Buckets](https://github.com/PaladinCloud/CE/wiki/AWS-Policy#deny-public-access-to-non-allow-listed-s3-buckets) 
If there is sensitive data on the S3 and it allows public access, that is immediately exploitable.

**High**

> A high-severity misconfiguration or vulnerability can be exploited by attackers with minimal interaction required from the user and can lead to a partial system compromise or data loss. While not as severe as critical vulnerabilities, high-severity issues should still be addressed as soon as possible to minimize the risk of a successful attack.

Example High Policy: [Deny HTTP Requests to S3 Bucket](https://github.com/PaladinCloud/CE/wiki/AWS-Policy#deny-http-requests-to-s3-bucket)
If there is sensitive data on the S3 and it allows HTTP requests, that isn't immediately exploitable. Still, it could be exploited with a little bit of social engineering or correct location and timing.

**Medium**

> A medium-severity vulnerability requires significant interaction from the user or an attacker and could lead to a limited system compromise or data loss. These vulnerabilities are often exploited in combination with other vulnerabilities to gain access to a system or network. While not as urgent, medium-severity issues should still be addressed in a timely manner.

Example High Policy: [Enable S3 Bucket Object-Level Logging for Read Events](https://github.com/PaladinCloud/CE/wiki/AWS-Policy#enable-s3-bucket-object-level-logging-for-read-events)
A lack of logging isn't directly exploitable, but it lets an attacker hide their actions and can make it so that even if discovered, what they did may not be fully identified after the fact.

**Low**

> A low-severity vulnerability is unlikely to be exploited by an attacker. However, a low misconfiguration may still allow an attacker to collect non-sensitive data or disrupt the functionality of a system or network. These violations should still be addressed to maintain security best practices. These vulnerabilities may not require immediate action. However, leaving low-severity issues unaddressed for an extended period can increase the risk of an attack.

Example Low Policy: [Enable MFA Delete on S3 Bucket](https://github.com/PaladinCloud/CE/wiki/AWS-Policy#enable-mfa-delete-on-s3-bucket)
This setting makes it so MFA is needed to delete an S3 bucket. This setting doesn't require MFA for read actions, so it doesn't add any security to stop a bad actor from getting and using the data. However, if they tried to delete the bucket, it might slow them down. AWS docs even say this is mainly useful for keeping users from mistakenly deleting an S3 bucket since an extra level of validation is needed. 

### Some of my thoughts
The four descriptions above make the most sense in a security sense. But they fall short for cost, operations, and tagging violations. For example, it probably doesn't make sense to evaluate a Cost violation using the same criteria for Security. 
Looking at existing policies, all Cost and Tagging policies are listed as high. Operations policies range from high to low. 

I'm not a big fan of defining these levels by saying a severity is less or more of X than an adjacent level. I do it in the description for high, but clearly defining that delineation would be nice.

What about you? Share your thoughts and any adjustments you would like made to these definitions.

**Comment by Steve-Hull** [comment]
The criteria make sense, but the descriptions are long.  One idea is to say “Critical Issues” instead of “Critical misconfigurations or vulnerabilities”.

Critical issues can lead to immediate exploitation of your system or network. Attackers can exploit these violations without user interaction, resulting in significant damage or data loss. Thus, addressing critical issues should be top priority and immediate.

**Comment by PaladinCloudJohn** [comment]
Good point. I've shortened the descriptions so they aren't as verbose.

### Violation Severity Definitions
**Critical**

Critical issues can lead to immediate exploitation of your system or network. Attackers can exploit these violations without user interaction, resulting in significant damage or data loss. Thus, addressing critical issues should be the top priority and immediate.

**High**

High issues can lead to the exploitation of your system or network. Attackers can exploit these violations with minimal user interaction, resulting in damage or data loss. Addressing high issues should be addressed as soon as possible.

**Medium**

Medium issues can lead to limited exploitation of your system or network. Exploiting these violations requires significant user interaction or a combination of vulnerabilities. Addressing medium issues should be done in a timely manner.

**Low**

Low issues may allow an attacker to collect non-sensitive data or disrupt the functionality of a system or network. While these issues may not require immediate action, leaving low-severity issues unaddressed for an extended period can increase the risk of an attack.
